shhext/service.go

```diff
@@ -0,0 +1,145 @@
+package shhext
+
+import (
+	"sync"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/node"
+	"github.com/ethereum/go-ethereum/p2p"
+	"github.com/ethereum/go-ethereum/rpc"
+	whisper "github.com/ethereum/go-ethereum/whisper/whisperv6"
+)
+
+// ConfirmationHandler used as a callback for confirming that envelopes were sent.
+type ConfirmationHandler func(common.Hash)
+
+// Service is a service that provides some additional Whisper API.
+type Service struct {
+	w       *whisper.Whisper
+	tracker *tracker
+}
+
+// Make sure that Service implements node.Service interface.
+var _ node.Service = (*Service)(nil)
+
+// New returns a new Service.
+func New(handler ConfirmationHandler, w *whisper.Whisper) *Service {
```

Comment:
I would swap `handler` with `w`. Handlers, callbacks etc. are usually passed as the last arguments.